cloud/scope/powervs_machine_test.go

```diff
@@ -190,6 +327,764 @@ func TestNewPowerVSMachineScope(t *testing.T) {
+			scope := PowerVSMachineScope{
+				IBMPowerVSClient: mockResourceController,
+			}
+			result, err := getNetworkID(networkResource, &scope)
+			require.Nil(t, err)
+			require.Equal(t, *result, expectedNetworkID)
+		})
+
+		t.Run("Failed to find network ID", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			expectedNetworkID := networkID
+			differentNetworkID := "diff-network-id"
+			networkReferences := []*models.NetworkReference{
+				{
+					Name: core.StringPtr(differentNetworkID),
+				},
+			}
+			networks := &models.Networks{
+				Networks: networkReferences,
+			}
+			networkResource := infrav1beta2.IBMPowerVSResourceReference{
+				Name: core.StringPtr(expectedNetworkID),
+			}
+
+			mockResourceController := mock.NewMockPowerVS(gomock.NewController(t))
+			mockResourceController.EXPECT().GetAllNetwork().Return(networks, nil)
+			scope := PowerVSMachineScope{
+				IBMPowerVSClient: mockResourceController,
+			}
+			result, err := getNetworkID(networkResource, &scope)
+			require.Nil(t, result)
+			require.EqualError(t, err, fmt.Sprintf("failed to find a network ID with name %s", expectedNetworkID))
+		})
+
+		t.Run("Fetch network ID with matching regex", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			networkName := "550e8400-e29b-41d4-a716-446655440000"
+			expectedNetworkID := "foo-id"
+			networkReferences := []*models.NetworkReference{
+				{
+					Name:      core.StringPtr(networkName),
+					NetworkID: core.StringPtr(expectedNetworkID),
+				},
+			}
+			networks := &models.Networks{
+				Networks: networkReferences,
+			}
+			networkResource := infrav1beta2.IBMPowerVSResourceReference{
+				RegEx: core.StringPtr("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"),
+			}
+
+			mockResourceController := mock.NewMockPowerVS(gomock.NewController(t))
+			mockResourceController.EXPECT().GetAllNetwork().Return(networks, nil)
+			scope := PowerVSMachineScope{
+				IBMPowerVSClient: mockResourceController,
+			}
+			result, err := getNetworkID(networkResource, &scope)
+			require.Equal(t, *result, expectedNetworkID)
+			require.Nil(t, err)
+		})
+
+		t.Run("Failed to fetch network ID with matching regex", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			expectedNetworkID := "foo-netID"
+			regex := "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+			networkReferences := []*models.NetworkReference{
+				{
+					Name: core.StringPtr(expectedNetworkID),
+				},
+			}
+			networks := &models.Networks{
+				Networks: networkReferences,
+			}
+			networkResource := infrav1beta2.IBMPowerVSResourceReference{
+				RegEx: core.StringPtr(regex),
+			}
+
+			mockResourceController := mock.NewMockPowerVS(gomock.NewController(t))
+			mockResourceController.EXPECT().GetAllNetwork().Return(networks, nil)
+			scope := PowerVSMachineScope{
+				IBMPowerVSClient: mockResourceController,
+			}
+			result, err := getNetworkID(networkResource, &scope)
```

Comment:
Across the code base I think we use gomega for assertion, Can we switch to that? something like

````
g := NewWithT(t)
g.Expect(err).To(BeNil())
```

Reply:
Will update this

Reply:
Done